### ISSUE TITLE
ValueObject custom attribute names

### ISSUE THREAD

**Issue by SerafimArts** [analogueorm/analogue#100]
I has 3 attribute in my database table:
- cost
- cost_month
- cost_mimial

Cost VO class like:

``` php
class Cost extends ValueObject
{
    public function __construct($connection, $minimal, $month)
    {
        $this->connection = $connection;
        $this->minimal = $minimal;
        $this->month = $month;
    }
}
```

And Mapper:

``` php
class CostMap extends ValueMap
{
    protected $attributes = [
        'connection',
        'minimal',
        'month'
    ];
}
```

How map `cost` attribute to `cost_connection` attribute? if that possibility.

If not - what about this syntax?

``` php
class CostMap extends ValueMap
{
    // `[int] => [attribute]` converts to `voname_[attribute]` => `[attribute]`
    // `[string] => [attribute]` converts to `[string] => [attribute]`
    protected $attributes = [
        'cost' => 'connection',
        'minimal',
        'month'
    ];
}
```

**Comment by RemiCollin** [comment]
Right now, the ValueObject's attributes are snakecase's namespaced , so your attributes `connection`, `minimal`, `month`, would map to database columns `cost_connection`, `cost_minimal`, `cost_month`. 

I'm looking for a more flexible mechanism though. The options I thought of so far are :

- Serializing all ValueObject to an array or mysql json column. 
- Be able to override the value object's attributes in the entityMap that uses the object (something like your example, but not on the ValueMap itself). 

Can be actually both. Have to think that through.

**Comment by Sufir** [comment]
Any progress?

**Comment by RemiCollin** [comment]
Not implemented yet, but working on a much improved & flexible ValueObject implementation for upcoming version 5.4 , which support these features.

**Comment by RemiCollin** [comment]
implemented in upcoming 5.5 release
